src/command_add.rs

```diff
@@ -110,5 +110,76 @@ fn add_non_db(channel: &str, paths: &GlobalPaths) -> Result<()> {
     if config_file.data.settings.create_channel_symlinks {
         create_symlink(&config_channel, &format!("julia-{}", channel), paths)?;
     }
+
+    // Handle codesigning for PR builds on macOS
+    #[cfg(target_os = "macos")]
+    if Regex::new(r"^pr\d+").unwrap().is_match(channel) {
+        if let Err(e) = codesign_pr_build_if_needed(channel, paths) {
+            eprintln!("\nWarning: Codesigning failed: {}", e);
+            eprintln!("The Julia binary may not run without manual codesigning.");
+        }
+    }
+
+    Ok(())
+}
+
+#[cfg(target_os = "macos")]
+fn codesign_pr_build_if_needed(channel: &str, paths: &GlobalPaths) -> Result<()> {
+    use std::io::{self, Write};
+
+    eprintln!("\nWARNING: PR builds are not code-signed for macOS.");
+    eprintln!("The Julia binary will fail to run unless you codesign it locally.");
```

Comment:
I know that this just works on other platforms, but I feel it should be spelled out given we're asking to work around the block.

```suggestion
    eprintln!("Note that unmerged PRs may not have been reviewed for security issues etc.");
    eprintln!("The Julia binary will fail to run unless you codesign it locally.");
```

Reply:
I guess we could move that message earlier when adding the channel, so it shows on all platforms.

Reply:
Well we aren't ad hoc codesigning on other platforms.

Reply:
> Well we aren't ad hoc codesigning on other platforms.

I don't see how that matters, the code runs the same. Having a common warning for all platforms makes sense, though.